aws_lambda_builders/workflows/dotnet_clipackage/DESIGN.md

```diff
@@ -0,0 +1,86 @@
+# .NET Core - Lambda Builder
+
+### Scope
+
+To build .NET Core Lambda functions this builder will use the AWS .NET Core Global Tool [Amazon.Lambda.Tools](https://github.com/aws/aws-extensions-for-dotnet-cli#aws-lambda-amazonlambdatools).
+This tool has several commands for building and publishing .NET Core Lambda functions. For this integration 
+the `dotnet lambda package` command will be used to create a zip file that can be deployed to Lambda.
+
+The builder will install the Amazon.Lambda.Tools Global Tool or update to the latest version before executing 
+the package command.
+
+This builder assumes the [.NET Core command-line interface (CLI)](https://docs.microsoft.com/en-us/dotnet/core/tools/?tabs=netcore2x) 
+is already installed and added to the path environment variable. This is a reasonable requirement as the 
+.NET Core CLI is a required tool for .NET Core developers to build any .NET Core project.
+
+The .NET Core CLI handles the validation that the correct version of .NET Core is installed and errors out when there is 
+not a correct version. 
+
+### Challenges
+
+#### Output
+
+The output of `dotnet lambda package` command is a zip archive that consumers can then deploy to Lambda. For SAM build
```

Comment:
Yes, this is a requirement mainly stemming from the ability to customize the zip file later during package. See this discussion - https://github.com/awslabs/aws-lambda-builders/pull/69#discussion_r256537324